### ISSUE TITLE
Moving from AWS to local inference, which GPU?

### ISSUE THREAD

**Issue by mehditlili** [matterport/Mask_RCNN#1262]
Hi everybody,
I have been using mask rcnn for a while now running on an AWS EC2 instance (p2.xlarge). This instance came with a Tesla k80 with half the memory (12Gb). It could run inference on 320x240 rgb images in 200ms. Now I am looking to have my own inference machine locally and am thinking about buying a GPU to replace the K80. I couldn't find any benchmarking for maskrcnn. This [blog post](http://timdettmers.com/2018/11/05/which-gpu-for-deep-learning/) was helpful but I wanted to ask you guys about your experience. Would an RTX 2070 be able to handle inference in 200ms, or even lower? I can't run inference on more than one image as I am getting data from a camera so my batch size in inference is limited to 1. Also do you have any experience with 16bit floats for maskrcnn?

**Comment by qzqzzz** [comment]
@mehditlili could you please share what is your environment setup:
- tensorflow-gpu version (pip3 show tensorflow-gpu)
- keras version
- scikit-image
- python version (python3 --version)
- cuda version (nvcc --version)
- cudnn version (cat /usr/local/cuda/include/cudnn.h | grep CUDNN_MAJOR -A 2)

I'm running inference on a Google Cloud n1-standard-4 google cloud VM with Tesla V100 GPU with 16GB (which is more powerful than a Tesla K80 GPU - see the comparison here https://developer.nvidia.com/cuda-gpus) and I'm getting somewhere around 3s / 320x320 picture, which is far behind what your achieved 200m / 320x240 picture.

I would say RTX 2070 is a good choice, but only for 16bit training (given that it has only 8GB). If you have extra cash, I would recommend going for a RTX 2080 Ti.

**Comment by phylliskaka** [comment]
@mehditlili Hi, I am also using Tesla K80 on AWS. I didnt get any problem when I training the head, but I encounter OOM when I start training all layers. My config is following: 
batch_size=4
resnet101
It take all the memory of GPU which is really odd. I am thinking is it because of the environment setup?

**Comment by mehditlili** [comment]
A small feedback, just ditch AWS and get your own GTX 1070, the training goes way faster, inference as well (100 ms VS 220 ms on AWS). I am really pissed at the few hundred bucks I've spent on AWS while a GTX 1070 costs around 300 dollars.

@phylliskaka I get this problem when I have a big batch, depending on your image size, you might have to use batch 2, or even batch 1. I also think it is odd and didn't have time to look closer into it but reducing the batch size might help you get going for now.
